apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1340,7 +1340,31 @@ func (u *UniterAPI) oneEnterScope(ctx context.Context, canAccess common.AuthFunc
 		return internalerrors.Capture(err)
 	}
 
-	err = u.relationService.EnterScope(ctx, relUUID, unitName)
+	unit, err := u.st.Unit(unitTag.Id())
+	if err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	addr, err := unit.PublicAddress()
+	if err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	settings := map[string]string{}
+	// private-address is historically a cloud local address for the machine.
+	// Existing charms are built to ask for this attribute from relation
+	// settings to find out what address to use to connect to the app
+	// on the other side of a relation. For cross model scenarios, we'll
+	// replace this with possibly a public address; we expect to fix more
+	// charms than we break - breakage will not occur for correctly written
+	// charms, since the semantics of this value dictates the use case described.
+	// Any other use goes against the intended purpose of this value.
+	settings["private-address"] = addr.String()
```

Comment:
We can omit this. We've given warning of its removal from 4.0:
https://discourse.charmhub.io/t/juju-relation-network-data/17179

Reply:
done